clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsTest.java

```diff
@@ -902,39 +898,39 @@ public void testNextBatchSize() {
         assertEquals(0, buffer.position());
 
         buffer.limit(1); // size not in buffer
-        assertEquals(null, records.firstBatchSize());
+        assertNull(records.firstBatchSize());
         buffer.limit(Records.LOG_OVERHEAD); // magic not in buffer
-        assertEquals(null, records.firstBatchSize());
+        assertNull(records.firstBatchSize());
         buffer.limit(Records.HEADER_SIZE_UP_TO_MAGIC); // payload not in buffer
         assertEquals(size, records.firstBatchSize().intValue());
 
         buffer.limit(size);
         byte magic = buffer.get(Records.MAGIC_OFFSET);
         buffer.put(Records.MAGIC_OFFSET, (byte) 10);
-        try {
-            records.firstBatchSize();
-            fail("Did not fail with corrupt magic");
-        } catch (CorruptRecordException e) {
-            // Expected exception
-        }
+        assertThrows(CorruptRecordException.class, records::firstBatchSize);
         buffer.put(Records.MAGIC_OFFSET, magic);
 
         buffer.put(Records.SIZE_OFFSET + 3, (byte) 0);
-        try {
-            records.firstBatchSize();
-            fail("Did not fail with corrupt size");
-        } catch (CorruptRecordException e) {
-            // Expected exception
-        }
+        assertThrows(CorruptRecordException.class, records::firstBatchSize);
     }
 
-    private void expectExceptionWithZStd(CompressionType compressionType, byte magic) {
-        if (compressionType == CompressionType.ZSTD && magic < MAGIC_VALUE_V2) {
-            exceptionRule.expect(IllegalArgumentException.class);
-            exceptionRule.expectMessage("ZStandard compression is not supported for magic " + magic);
+    @Test
+    public void testWithRecords() {
+        Supplier<MemoryRecords> recordsSupplier = () -> MemoryRecords.withRecords(magic, compression,
+            new SimpleRecord(10L, "key1".getBytes(), "value1".getBytes()));
+        if (compression != CompressionType.ZSTD || magic >= MAGIC_VALUE_V2) {
+            MemoryRecords memoryRecords = recordsSupplier.get();
+            String key = Utils.utf8(memoryRecords.batches().iterator().next().iterator().next().key());
+            assertEquals("key1", key);
+        } else {
+            assertThrows(IllegalArgumentException.class, recordsSupplier::get);
         }
     }
 
+    private void assumeV2OrNotZstd() {
+        assumeTrue(compression != CompressionType.ZSTD || magic == MAGIC_VALUE_V2);
```

Comment:
magic comparison to v2 has same equality change here as in the previous test. any reason not to `>=`?

Reply:
Sorry for missing this one, fixed.